src/lib/vm/vm.js

```diff
@@ -2069,16 +2070,24 @@ export default class VM {
   }
 
   vmRequire(src) {
-    const [srcPath, version] = src.split("@");
-    const code = this.cachedSocialGet(
-      srcPath.toString(),
-      false,
-      version, // may be undefined, meaning `latest`
-      undefined
-    );
-    if (!code) {
-      return code;
+    const srcOrCode = computeSrcOrCode(src, null, this.widgetConfigs);
+    let code;
+    if (srcOrCode?.src) {
+      const src = srcOrCode.src;
+      const [srcPath, version] = src.split("@");
+      code = this.cachedSocialGet(
+        srcPath.toString(),
+        false,
+        version, // may be undefined, meaning `latest`
+        undefined
+      );
+      if (!code) {
+        return code;
+      }
+    } else if (srcOrCode?.code) {
+      code = srcOrCode.code;
     }
+    console.log(code);
     const vm = this.getVmInstance(code, src);
```

Comment:
This should be updated based on `srcOrCode?.src`
```suggestion
    const vm = this.getVmInstance(code, srcOrCode?.src);
```